pkgs/by-name/ca/catppuccin-plymouth/package.nix

```
@@ -0,0 +1,56 @@
+{
+  stdenvNoCC,
+  lib,
+  fetchFromGitHub,
+  unstableGitUpdater,
+  variant ? "macchiato",
+}:
+
+let
+  pname = "catppuccin-plymouth";
+  validVariants = [
+    "latte"
+    "frappe"
+    "macchiato"
+    "mocha"
+  ];
+in
+lib.checkListOfEnum "${pname}: color variant" validVariants [ variant ]
+
+  stdenvNoCC.mkDerivation
+  (finalAttrs: {
+    inherit pname;
+    version = "0-unstable-2024-05-28";
+
+    src = fetchFromGitHub {
+      owner = "catppuccin";
+      repo = "plymouth";
+      rev = "e13c348a0f47772303b2da1e9396027d8cda160d";
+      hash = "sha256-6DliqhRncvdPuKzL9LJec3PJWmK/jo9BrrML7g6YcH0=";
+    };
+
+    sourceRoot = "${finalAttrs.src.name}/themes/catppuccin-${variant}";
+
+    installPhase = ''
+      runHook preInstall
+
+      sed -i 's:\(^ImageDir=\)/usr:\1'"$out"':' catppuccin-${variant}.plymouth
+      mkdir -p $out/share/plymouth/themes/catppuccin-${variant}
+      cp * $out/share/plymouth/themes/catppuccin-${variant}
+
+      runHook postInstall
+    '';
+
+    passthru.updateScript = unstableGitUpdater { hardcodeZeroVersion = true; };
```

The script can infer on its own that it needs to use `0` if no previous tags exist. There's a hint in the script's comments about this being necessary, but that's bitrotten & doesn't apply for tag-less repos anymore. This shouldn't be necessary.

```suggestion
    passthru.updateScript = unstableGitUpdater { };
```

(I'll fix the documentation bitrot ASAP.)

---

Done. Thanks.